Dockerfiles/agent-ddot/Dockerfile.agent-otel

```diff
@@ -27,19 +27,16 @@ RUN apt-get update && \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
-# TEMP: Use github source code
-RUN git clone --filter=blob:none --branch "${AGENT_BRANCH}" --single-branch https://github.com/DataDog/datadog-agent.git datadog-agent-${AGENT_VERSION}
-
 # Once we have stable releases, we can use the following code to download the source code
 # TODO: use released agent version once we have an agent release with the otel binary
 # Download and unpack source code
-#RUN curl -L https://github.com/DataDog/datadog-agent/archive/refs/tags/${AGENT_VERSION}.tar.gz -o datadog-agent-${AGENT_VERSION}.tar.gz && \
-#    tar -xzvf datadog-agent-${AGENT_VERSION}.tar.gz && \
-#    rm datadog-agent-${AGENT_VERSION}.tar.gz
+RUN curl -L https://github.com/DataDog/datadog-agent/archive/refs/tags/${AGENT_BRANCH}.tar.gz -o datadog-agent-${AGENT_BRANCH}.tar.gz && \
```

Comment:
This seems to be introducing issue:
> ERROR: process "/bin/sh -c curl -L https://github.com/DataDog/datadog-agent/archive/refs/tags/${AGENT_BRANCH}.tar.gz -o datadog-agent-${AGENT_BRANCH}.tar.gz &&     tar -xzvf datadog-agent-${AGENT_BRANCH}.tar.gz &&     rm datadog-agent-${AGENT_BRANCH}.tar.gz" did not complete successfully: exit code: 23

https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/945724003